### ISSUE TITLE
🐛 Vue parser fails with `@keydown.:.stop.prevent="..."`

### ISSUE THREAD

**Issue by Blechlawine** [biomejs/biome#8765]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-kitty
  JS_RUNTIME_VERSION:           v24.11.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.4

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### What happened?

1. use this attribute in a vue file: `@keydown.:.stop.prevent="() => {}"`
2. I think the `:` part is probably the problem, but this is valid vue syntax and works at runtime as expected
3. the output i get is:
```bash
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /__w/biome/biome/crates/biome_parser/src/diagnostic.rs:395:14
Thread Name: biome::workspace_worker_8
Message: Expected token to be a punctuation or keyword.

admin/src/components/input/Time.vue internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Expected token to be a punctuation or keyword.

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
```
In the referenced file, when i removed that attribute, biome didn't panic, but formatted everything as expected

### Expected result

it should not panic

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Hey @Blechlawine 

We need a few things:
- a reproduction (you can use the playground I think)
- the Vue docs that explain this as a valid syntax
- re-run the command by preceding `RUST_BACKTRACE=1` to the Biome command you use, and past here the stacktrace of the diagnostic

**Comment by Blechlawine** [comment]
running the biome command with RUST_BACKTRACE=1 doesn't print anything more, just the same output.
the biome command I'm running:
```bash
bun biome format --write .
```

here is a minimal repro on the biome playground, i don't know if the missing output from the biome formatter is related to the problem: https://biomejs.dev/playground/?code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AGkAbgBwAHUAdAAgAEAAawBlAHkAZABvAHcAbgAuADoALgBzAHQAbwBwAC4AcAByAGUAdgBlAG4AdAA9ACIAZgBvAGMAdQBzAE0AaQBuAHUAdABlAEkAbgBwAHUAdAAiAD4ACgA8AC8AdABlAG0AcABsAGEAdABlAD4ACgA%3D&language=vue

this is the link to the relevant vue docs: https://vuejs.org/guide/essentials/event-handling.html#key-modifiers

**Comment by dyc3** [comment]
This is a bug in that biome's parser should emit a diagnostic and not panic, not because its valid syntax. `.:` is not a valid modifier.

**Comment by Blechlawine** [comment]
@dyc3 The Vue docs mentionend state this:
> You can directly use any valid key names exposed via [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values) as modifiers by converting them to kebab-case.

and ":" is indeed a valid value for `KeyboardEvent.key` (you can check yourself with this tool: https://www.toptal.com/developers/keycode) so why should this not be valid syntax?

It also works as one would expect at runtime and vue itself doesn't show any errors or warnings.

**Comment by dyc3** [comment]
Ah, I see, Thanks for clarifying.
